src/js/moza.js

```diff
@@ -218,29 +248,41 @@
   };
 
   /*
-  * foundTileSize
+  * Get next tile size
+  * This will get the next tile size to use.
+  * @param {String} tileCount
   */
-  Tiles.prototype.foundTileSize = function(tileCount) {
-    var size = 'small';
-    if(tileCount < constants.TILE_SIZE.big.maxAmount) {
-      size = 'big';
-    }else if(tileCount < constants.TILE_SIZE.big.maxAmount + constants.TILE_SIZE.medium.maxAmount) {
-      size = 'medium';
+  Tiles.prototype.getNextTileSize = function(tileIndex) {
+    console.log('getNextTileSize');
+    var currentTileCount = 0;
+    var tileSize = null;
+    for(let size in constants.TILE_SIZE){
+      currentTileCount = currentTileCount + constants.TILE_SIZE[size].maxAmount;
+      if(tileIndex <= currentTileCount){
+        return size;
+      }
     }
-    return size;
+    return tileSize;
   };
 
-
   /*
-  * reduceTileSize
+  * Reduce tile size
+  * This is checking all the tile size and look for the next area smaller then the current one.
+  * To find the area we just ened to multiply the col and row (constants.TILE_SIZE[size].col * constants.TILE_SIZE[size].row)
+  * @param {string} currentTileSize - big, medium, small, ect.
   */
-  Tiles.prototype.reduceTileSize = function(currentSize) {
-    switch(currentSize) {
-      case 'big':
-        return 'medium';
-      case 'medium':
-        return 'small';
+  Tiles.prototype.reduceTileSize = function(currentTileSize) {
+    console.log('reduceTileSize');
+    var currentTile = constants.TILE_SIZE[currentTileSize];
+    var currentTileArea = currentTile.col * currentTile.row;
+    var nextSize = null; // This will return null if no smaller tile are found.
+    for (let size in constants.TILE_SIZE) {
+      let nextTileArea = constants.TILE_SIZE[size].col * constants.TILE_SIZE[size].row;
+      if (nextTileArea < currentTileArea) {
+        return size;
+      }
     }
+    return nextSize;
   };
 
   /*
```

Comment:
All method description should be consistent. Copy/past function name as a description is baaddddd!